src/Clock.tsx

```diff
@@ -0,0 +1,54 @@
+/* eslint-disable no-console */
+import React from 'react';
+
+type Props = {
+  clockName: string;
+};
+
+type State = {
+  today: Date;
+};
+
+export class Clock extends React.Component<Props, State> {
+  state = {
+    today: new Date(),
+  };
+
+  timerId = 0;
+
+  componentDidMount(): void {
+    this.timerId = window.setInterval(() => {
+      this.setState({ today: new Date() });
+      console.info(this.state.today.toUTCString().slice(-12, -4));
+    }, 1000);
+  }
+
+  componentDidUpdate(prevProps: Readonly<Props>): void {
+    if (prevProps.clockName !== this.props.clockName) {
+      console.debug(`Renamed from ${prevProps.clockName} to ${this.props.clockName}`);
+    }
+  }
+
+  componentWillUnmount(): void {
+    window.clearInterval(this.timerId);
+  }
+
+  render() {
+    const { today } = this.state;
+    const { clockName } = this.props;
+
+    return (
+      <div className="Clock">
+        <strong className="Clock__name">
+          {clockName}
+        </strong>
+
+        {' time is '}
+
+        <span className="Clock__time">
+          {today.toUTCString().slice(-12, -4)}
```

Comment:
It's better to not calculate this value inside JSX. The JSX code should be clean and focused on rendering components rather than defining complex logic directly within it